### PR TITLE
Upgrade Prometheus to v0.41.0

### PR DESCRIPTION
**Pull request by rabenhorst**
Signed-off-by: Sebastian Rabenhorst <sebastian.rabenhorst@shopify.com>

Upgrade Prometheus to version v0.41.0 and fixed binary reader so it works with optimized posting offset table reading (https://github.com/prometheus/prometheus/pull/11535).

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

- Upgrade Prometheus to `v0.41.0`

### CODE REVIEW

File: pkg/block/indexheader/binary_reader.go [hunk 1]
```diff
@@ -520,84 +520,79 @@ func newFileBinaryReader(path string, postingOffsetsInMemSampling int) (bw *Bina
...
 	if r.indexVersion == index.FormatV1 {
 		// Earlier V1 formats don't have a sorted postings offset table, so
 		// load the whole offset table into memory.
 		r.postingsV1 = map[string]map[string]index.Range{}
 
 		var prevRng index.Range
-		if err := index.ReadOffsetTable(r.b, r.toc.PostingsOffsetTable, func(key []string, off uint64, _ int) error {
-			if len(key) != 2 {
-				return errors.Errorf("unexpected key length for posting table %d", len(key))
+		if err := index.ReadPostingsOffsetTable(r.b, r.toc.PostingsOffsetTable, func(name, value []byte, postingsOffset uint64, _ int) error {
```
Comment (fpetkovski) [inline pkg/block/indexheader/binary_reader.go:530]: Is this taken from Prometheus?

Comment (rabenhorst) [reply pkg/block/indexheader/binary_reader.go:530]: It's "inspired" by this: https://github.com/prometheus/prometheus/blob/3b4cbf8da4c5b9b8c6c2a0a30ae182dfbb0b8213/tsdb/index/index.go#L1166-L1217

**Review by fpetkovski — APPROVED** [review]

**Review by yeya24 — APPROVED** [review]
Thanks. Let's fix the conflicts and we can merge the pr.

**Comment by rabenhorst** [comment]
> Thanks. Let's fix the conflicts and we can merge the pr.

Done.

**Review by fpetkovski — APPROVED** [review]

**Review by saswatamcode — APPROVED** [review]

**Pull request merged**